### ISSUE TITLE
meta.nodeNixpkgs

### ISSUE THREAD

**Issue by blaggacao** [zhaofengli/colmena#71]
It is really a nightmare in terms of composability to have this attribute exposed via `meta`. I wonder if it would be possible instead to expose it via the normal (composable) module system and shallow evaluate for peek. something like this:

```nix
let
  colmenaNixpkgs = {
    options.colmenaNixpkgs = {};
  };
in (nixpkgs.lib.evalModules { modules = [colmenaNixpkgs]; check = false; }).colmenaNixpkgs
```

**Comment by zhaofengli** [comment]
Yeah, I agree that it doesn't lend to composability. Having it as part of the module system is a bit tricky to get right, and #56 was one attempt at doing do.

**Comment by blaggacao** [comment]
Some discussion participant in https://github.com/zhaofengli/colmena/issues/55 apparently chose to behave exclusive towards me on the other thread, so I technically can't comment there at this time.

But I think even the use of `defaults` & `meta` might be questionable. This introduces magic on a principal level where nix, the language would be powerful & expressive enough so that users &mdash; for the sake of transparency &mdash; wouldn't be dependent on that magic.

**Comment by blaggacao** [comment]
I think I'll just copy&adapt https://github.com/NixOS/nixops/pull/1508

**Comment by blaggacao** [comment]
I discussed with @roberth and we kind of concluded that upstream `nixos` should eventually implement `nixpkgs.source` in addition to all the other `nixpkgs.*` options.

This makes a lot of sense and would allow us to fully configure `nixpkgs` at the leaves of the hive-configs and completely avoid `nodeNixpkgs` - we could do a shallow pass on the configurations by just using the upstream `nixpkgs` module (or a fall-back for older nixpkgs versions) and then evaluate the configuration with these options.

Another, maybe even cleaner option is not to confuse the different `nixpkgs` (lib, nixos, pkgs) and instead make it a restriction of colmena to at least define:

```nix
{
  nixpkgs.source = "...";
  nixpkgs.system = "x86_64-linux";
}
```

That should _almost_ suffice completely, if it wasn't for a nixos bug where the configuration hard-coded depends on the current evaluator's platform: 
```nix
      # Stash the value of the `system` argument. When using `nesting.children`
      # we want to have the same default value behavior (immediately above)
      # without any interference from the user's configuration.
      nixpkgs.initialSystem = system;
```
